src/lib/memory/MMU.cc

```diff
@@ -0,0 +1,36 @@
+#include "simeng/memory/MMU.hh"
+
+#include "simeng/OS/Constants.hh"
+namespace simeng {
+namespace memory {
+
+MMU::MMU(std::shared_ptr<Mem> memory, VAddrTranslator fn, uint64_t tid) {
+  memory_ = memory;
+  translate_ = fn;
+  tid_ = tid;
+};
+
+void MMU::bufferRequest(DataPacket request, sendResponseToCore sendResponse) {
+  // Since we don't have a TLB yet, treat every memory request as a TLB miss and
+  // consult the page table.
+  uint64_t paddr = translate_(request.address_, tid_);
+  uint64_t faultCode = simeng::OS::masks::faults::getFaultCode(paddr);
+  DataPacket pkt = DataPacket();
```

Comment:
Given pkt in all cases will be constructed anyway, you can just define it without initialising it : `DataPacket pkt;`